File: Python/Variables/variable_more.py
x = 5
num = 7.25

def main():
    a = 10
    b = 3.14
    c = 'A'

    d = c
    ptr = c

    vb = True  # true = 1, false = 0
    vc = False

    # print variables
    print("Variable values:")
    print(f"Value of a: {a}")
    print(f"Value of b: {b}")
    print(f"Value of c: {c}")
    print(f"Value of d: {d}")
    print(f"Value of *ptr: {ptr}")
    print(f"Value of vb: {int(vb)}")
    print(f"Value of vc: {int(vc)}")

    # print addresses
    print("\nVariable addresses:")
    print(f"Address of a: {id(a)}")
    print(f"Address of b: {id(b)}")
    print(f"Address of c: {id(c)}")
    print(f"Address of d: {id(d)}")
    print(f"Address of ptr: {id(ptr)}")

    c = chr(66)
    print(f"\nValue of c: {c}")
    ptr = d
    print(f"Value of *ptr: {ptr}")
    print(f"Address of ptr: {id(ptr)}")

    def rue():
        w = 70
        print(f"\nValue of w: {w}")
        print(f"Value of x: {x}")

        def sue():
            print(f"Value of num: {num}")
            print(f"Sum of num+w: {num + w:.2f}")
        
        sue()
    
    rue()
    
    # print(f"\nValue of w: {w}") # Error: w is not defined

def test():
    print(f"Sum of x+num: {x+num:.2f}")
    # print(f"Value of c: {c}") # Error: c is not defined

if __name__ == "__main__":
    main()

File: Cpp/Variables/variable_more.cpp
#include <iostream>
#include <iomanip>
#include <stdbool.h>

#define num 7.25

int x = 5;

int main() {
    int a = 10;
    float b = 3.14;
    char c = 'A';

    char d = c;
    char *ptr = &c;

    bool vb = true; // true = 1, false = 0
    bool vc = false;

    // print variables
    std::cout << "Variable values:" << std::endl;
    std::cout << "Value of a: " << a << std::endl;
    std::cout << "Value of b: " << b << std::endl;
    std::cout << "Value of c: " << c << std::endl;
    std::cout << "Value of d: " << d << std::endl;
    std::cout << "Value of *ptr: " << *ptr << std::endl;
    std::cout << "Value of vb: " << vb << std::endl;
    std::cout << "Value of vc: " << vc << std::endl;

    // print addresses
    std::cout << "\nVariable addresses:" << std::endl;
    std::cout << "Address of a: " << &a << std::endl;
    std::cout << "Address of b: " << &b << std::endl;
    std::cout << "Address of c: " << static_cast<void*>(&c) << std::endl;
    std::cout << "Address of d: " << static_cast<void*>(&d) << std::endl;
    std::cout << "Address of ptr: " << static_cast<void*>(ptr) << std::endl;

    *ptr = 66;
    std::cout << "\nValue of c: " << c << std::endl;
    ptr = &d;
    std::cout << "Value of *ptr: " << *ptr << std::endl;
    std::cout << "Address of ptr: " << static_cast<void*>(ptr) << std::endl;

    {
        int w = 70;
        std::cout << "\nValue of w: " << w << std::endl;
        std::cout << "Value of x: " << x << std::endl;

        {
            std::cout << "Value of num: " << num << std::endl;
            std::cout << "Sum of num+w: " << std::fixed << std::setprecision(2) << num + w << std::endl;
        }
    }

    return 0;
}